addons/dockable_container/dockable_container.gd

```diff
@@ -195,14 +253,24 @@ func set_layout(value: DockableLayout) -> void:
 		_layout.changed.disconnect(queue_sort)
 	_layout = value
 	_layout.changed.connect(queue_sort)
+	for window in _windows_container.get_children():
+		if not window.name in _layout.windows and window is FloatingWindow:
+			window.prevent_data_erasure = true  # We don't want to delete data.
+			window.close_requested.emit()  # Removes the window.
```

Comment:
Instead of using `prevent_data_erasure`, closing all windows and reopening them later, we could use a pool of FloatingWindow nodes, much like we pool DockablePanel and SplitHandle nodes. In `_resort`, we would call `_untrack_children_after(_windows_container)` after setting all active floating windows, which would delete unused windows, making them close anyway.